runelite-client/src/main/java/net/runelite/client/plugins/blastfurnace/BlastFurnacePlugin.java

```diff
@@ -135,4 +158,27 @@ public void onGameStateChanged(GameStateChanged event)
 			barDispenser = null;
 		}
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick tick)
+	{
+		Widget npcDialog = client.getWidget(WidgetInfo.DIALOG_NPC_TEXT);
+
+		if (foremanCounter != null && foremanCounter > 0)
+		{
+			foremanCounter--;
```

Comment:
Rather than measuring in ticks, save the time at which the foreman permission text is last found, then compute the time remaining in the overlay's render function. The time left to display should be the static duration (10 minutes) minus the difference between that saved timestamp and the current time.

Reply:
Resolved in 0fadcde